src/jit/lsra.cpp

```diff
@@ -10783,7 +10783,23 @@ void LinearScan::resolveEdge(BasicBlock*      fromBlock,
         regNumber targetReg = genRegNumFromMask(targetRegMask);
         if (location[targetReg] == REG_NA)
         {
-            targetRegsReady |= targetRegMask;
+#ifdef _TARGET_ARM_
+            regNumber sourceReg = (regNumber)source[targetReg];
+            Interval* interval = sourceIntervals[sourceReg];
+            if (interval->registerType == TYP_DOUBLE)
+            {
+                // For ARM32, make sure that two float reigster
+                // consisting a double register are not used.
+                assert(genIsValidDoubleReg(targetReg));
+                regNumber anotherHalfRegNum = REG_NEXT(targetReg);
+                if (location[anotherHalfRegNum] == REG_NA)
+                {
+                    targetRegsReady |= targetRegMask;
+                }
+            }
+            else
+#endif // _TARGET_ARM_
+                targetRegsReady |= targetRegMask;
```

Comment:
Is this right place to add logic to handle TYP_DOUBLE ?
Because I assumes that `targetRegsReady` contains regMask of free registers, I added code to consider two float registers here.

Reply:
I see. Thanks.